babel/src/board.js

```diff
@@ -284,6 +284,18 @@ export class Board {
         return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
 
+    swap(swapOrbs) {
+        let [[x1, y1], [x2, y2]] = swapOrbs;
+        let orbsBefore = _.cloneDeep(this.orbs);
```

Comment:
Yes, here, you're referring back to the original `orbsBefore` in case the swap yields no new matches, so you'll need a fully copy to keep as a backup in case that happens.

Other times though, there's no need to make a deep clone like this.